ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/result-filter/index.tsx

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+import * as React from 'react'
+import { hot } from 'react-hot-loader'
+import styled from '../../styles/styled-components'
+
+import { Button, buttonTypeEnum } from '../button'
+
+import MarionetteRegionContainer from '../../container/marionette-region-container'
+import * as FilterBuilderView from '../../../component/filter-builder/filter-builder.view'
+import * as FilterBuilderModel from '../../../component/filter-builder/filter-builder'
+
+type Props = {
+  removeFilter: () => void
+  saveFilter: (transformToCql: () => any) => void
+  hasFilter: Boolean
+  resultFilter?: {}
+  isList?: Boolean
+}
+
+const Root = styled<Props, 'div'>('div')`
+  display: block;
+  padding: ${props => props.theme.minimumSpacing};
+
+  .editor-footer {
```

Comment:
We should move these selector-based styles out into their own styled blocks and use those in the JSX definition.